bpzpipe/BPZPipe2.py

```diff
@@ -0,0 +1,351 @@
+
+        # Create the space for output data
+        nz = len(z)
+        group = f.create_group('pdf')
+        group.create_dataset("z", (nz,), dtype='f4')
+        group.create_dataset("pdf", (nobj,nz), dtype='f4')
+        group.create_dataset("mu", (nobj,), dtype='f4')
+        group.create_dataset("mu_1p", (nobj,), dtype='f4')
+        group.create_dataset("mu_1m", (nobj,), dtype='f4')
+        group.create_dataset("mu_2p", (nobj,), dtype='f4')
+        group.create_dataset("mu_2m", (nobj,), dtype='f4')
+
+        # One processor writes the redshift axis to output.
+        if self.rank==0:
+            group['z'][:] = z
+
+        return f
+
+    def load_templates(self):
+        from useful_py3 import get_str, get_data, match_resol
+
+
+        # The reshift range we will evaluate on
+        zmin = self.config['zmin']
+        zmax = self.config['zmax']
+        dz = self.config['dz']
+        z = np.arange(zmin, zmax + dz, dz)
+
+
+        bpz_path = self.config['path_to_bpz']
+        columns_file = '/Users/jaz/src/BPZpipe/test/CSDC2114_test.columns'
+        ignore_rows =['M_0','OTHER','ID','Z_S']
+        filters = [f for f in get_str(columns_file, 0) if f not in ignore_rows]
+
+        m0_index = 'ugrizy'.index('i')
+        spectra_file = os.path.join(bpz_path, self.config['spectra_file'])
+        spectra = [s[:-4] for s in get_str(spectra_file)]
+        ab_dir = os.path.join(bpz_path, self.config['ab_dir'])
+
+        nt = len(spectra)
+        nf = len(filters)
+        nz = len(z)
+        flux_templates = np.zeros((nz,nt,nf))
+
+        for i,s in enumerate(spectra):
+            for j,f in enumerate(filters):
+                model = f"{s}.{f}.AB"
+                model_path=os.path.join(ab_dir, model)
+                zo, f_mod_0 = get_data(model_path,(0,1))
+                flux_templates[:,i,j] = match_resol(zo, f_mod_0, z)
+
+        return z, flux_templates
+
+    def preprocess_magnitudes(self, data):
+        from bpz_tools_py3 import mag2flux, e_mag2frac
+
+        bands = self.config['bands']
+        suffices = ["", "_1p", "_1m", "_2p", "_2m"]
+
+        # Load the magnitudes
+        zp_errors = np.array(self.config['zp_errors'])
+        zp_frac=e_mag2frac(zp_errors)
+
+        # Only one set of mag errors
+        mag_errs = np.array([data[f'mag_err_{b}_lsst'] for b in bands]).T
+
+        # But many sets of mags, for now
+        for suffix in suffices:
+            # Group the magnitudes and errors into one big array
+            mags = np.array([data[f'mag_{b}_lsst{suffix}'] for b in bands]).T
+
+            # Clip to min mag errors
+            np.clip(mag_errs, self.config['mag_err_min'], 1e10, mag_errs)
+
+            # Convert to pseudo-fluxes
+            flux = 10.0**(-0.4*mags)
+            flux_err = flux * (10.0**(0.4*mag_errs) - 1.0)
+
+            # Check if an object is seen in each band at all.
+            # Fluxes not seen at all are listed as infinity in the input,
```

Comment:
Fluxes are not listed as infinity in the input, the convention that BPZ follows is that non-detections have a reported magnitude of 99.0, and in place of a magnitude error the mag_error column should contain the 1-sigma flux limit.  BPZ then replaces the flux for non-detections with zero, and the flux-error with the 1-sigma limit converted to flux

Reply:
What I meant to say here is that the input photometry_catalog.hdf files that I'm mocking currently use infinity for undetected.  I'll add an option to make this work for both cases.

Although it looks like I'm currently setting mag_err=inf for undetected objects, which clearly makes no sense, so I'll need to deal with that as well.